ruby/lib/google/protobuf/ffi/descriptor.rb

```diff
@@ -106,6 +106,15 @@ def options
         end
       end
 
+      def to_proto
```

Comment:
I think we should follow the C++ convention of naming these something like "copy_to".  I think it's an important distinction that these aren't necessarily 100% preserved in a roundtrip.  They also unfortunately reach into a lot of our internals (that we usually want to hide), so I think this alternate naming will help discourage use a bit.

Reply:
Thanks for your review! 
You mean, `#copy_to_proto` or just `#copy_to`?

In my personal opinion, using `#to_xx` is natural for generating another object from an object in Ruby's context(e.g. `Hash#to_a`). Sometimes those methods are costable. But, at Ruby, I think understandable naming is preferred.
But I would like to follow the Protocol Buffers's rule. So let me clarify the name.

Reply:
Well tbh we haven't standardized this yet.  In C++ we use `copyTo(DescriptorProto* out)`, and in Java we use `DescriptorProto toProto()` (and python doesn't offer this at all).  The Java implementation doesn't copy, so the name seems accurate but it makes it *really* easy to reach into our internals.  For example, we try to hide raw editions features because they're easy to misuse, but in Java you can simply call `toProto().getOptions().getFeatures()` whenever you want because we need to preserve round-trip behavior.

If `to_proto` is really more ergonomic for Ruby I guess it's fine, it's just kindof unfortunate

Reply:
Thanks for your detailed explanation! 
I understand your concern. But, I still `#to_proto` is better for Ruby. And, we already have `#to_proto` for the Message class in Ruby.  
https://github.com/protocolbuffers/protobuf/blob/ae3015c78a0ea3baa76f3ad556c0ecd575f2e618/ruby/lib/google/protobuf/message_exts.rb#L24-L26

So if we choose `#copy_to`,  it may cause confusion. So if it's OK, I would like to keep this name :pray:.